contrib/build_compat_tables.py

```diff
@@ -61,20 +61,28 @@
     :width: 32
 """
 
-# Credit http://stackoverflow.com/questions/42195468/how-can-i-inspect-an-attribute-of-a-module-without-importing-it?noredirect=1#comment71553012_42195468
+# Credit http://stackoverflow.com/questions/42195468/
+#        how-can-i-inspect-an-attribute-of-a-module-without-importing-it?noredirect=1#comment71553012_42195468
+
+
 def get_declaration_from_source(text, name="__all__"):
     tree = ast.parse(text)
     for node in tree.body:
-        if isinstance(node, ast.Assign) and len(node.targets)==1:
+        if isinstance(node, ast.Assign) and len(node.targets) == 1:
             target = node.targets[0]
             if isinstance(target, ast.Name) and target.id == name:
                 return ast.literal_eval(node.value)
-    raise NameError("name %r was not found"%(name,))
+    raise NameError("name %r was not found" % (name, ))
+
 
 _maps = dict.fromkeys(COMPONENT_LIST)
 
 for module, label in PLATFORM_LIST.items():
-    path = os.path.join('../src', module, 'toga_'+module, '__init__.py')
+    factory_path = os.path.join('../src', module, 'toga_'+module, 'factory.py')
+    if os.path.exists(factory_path):
+        path = factory_path
+    else:
+        path = os.path.join('../src', module, 'toga_'+module, '__init__.py')
```

Comment:
The spacing around the `+` is a little odd here (I'm surprised it isn't picked up by flake8...)

Reply:
I would suggest using `pathlib` in place of `os.path` as a whole. The API is considerably more Pythonic and easier to reason for most newcomers to the codebase in my opinion.

Reply:
Sure - pathlib is nice, but the *rest* of the codebase isn't using pathlib. Switching directory path mechanisms is a worthwhile change, but it's orthogonal to *this* change, so including that change here wouldn't be appropriate.

A *separate* pull request that ports path usage to pathlib would be welcome, though.

Reply:
Perhaps I could *start* said merge request, but I may only be able to provide commits for certain backends that I can personally test. I would have to rely on other maintainers to provide the commits and test results for backends like `cocoa` and `iOS`. Aside from that I would be happy to create a PR for this.

Reply:
I will ensconce that `+` in spacing today. Thanks!